docs/design/014-did-doc/README.md

```diff
@@ -0,0 +1,590 @@
+//
+// ddoc_builder_handle: builder handle
+// id: identifier of key
+// key_json: key_json in format compatible with did method
+// returns - ():
+pub fn indy_did_doc_update_authorization(ddoc_builder_handle,
+                                         id,
+                                         Option<key_json>) -> Future<()>
+```
+
+#### Delete authorization item with DID Doc builder
+
+```Rust
+// Delete authorization item with DID Doc builder.
+//
+// Note: id can be full DID Doc reference or fragment.
+// In case of fragment reference will be related to current DID.
+//
+// ddoc_builder_handle: builder handle
+// id: identifier of key
+// returns - ():
+pub fn indy_did_doc_remove_authorization(ddoc_builder_handle,
+                                         id) -> Future<()>
+```
+
+#### Add new service endpoint item to DID Doc builder
+
+```Rust
+// Add new service endpoint item to DID Doc builder.
+//
+// Note: id can be full DID Doc reference or path.
+// In case of path reference will be related to current DID.
+//
+// ddoc_builder_handle: builder handle
+// id: identifier of service endpoint
+// endpoint_json: endpoint_json in format compatible with did method.
+// returns - ():
+pub fn indy_did_doc_add_service_endpoint(ddoc_builder_handle,
+                                         id,
+                                         endpoint_json) -> Future<()>
+```
+
+#### Update service endpoint item with DID Doc builder
+
+```Rust
+// Update service endpoint item with DID Doc builder.
+//
+// Note: id can be full DID Doc reference or path .
+// In case of path reference will be related to current DID.
+//
+// ddoc_builder_handle: builder handle
+// id: identifier of key
+// endpoint_json: endpoint_json in format compatible with did method.
+// returns - ():
+pub fn indy_did_doc_update_service_endpoint(ddoc_builder_handle,
+                                            id,
+                                            endpoint_json) -> Future<()>
+```
+
+#### Delete service endpoint item with DID Doc builder
+
+```Rust
+// Delete service endpoint item with DID Doc builder.
+//
+// Note: id can be full DID Doc reference or path.
+// In case of path reference will be related to current DID.
+//
+// ddoc_builder_handle: builder handle
+// id: identifier of key
+// returns - ():
+pub fn indy_did_doc_remove_service_endpoint(ddoc_builder_handle,
+                                            id) -> Future<()>
+```
+
+#### Finalize DID Doc builder and return corresonded DID Doc or DID Doc delta
+
+```Rust
+// Finalize DID Doc builder and return corresonded DID Doc or DID Doc delta.
+//
+// Note: Format of delta can be method specific.
+// In case of path reference will be related to current DID.
+//
+// ddoc_builder_handle: builder handle
+// id: identifier of key
+// returns - did_doc_delta: DID Doc or DID Doc delta
+pub fn indy_did_doc_finalize(ddoc_builder_handle) -> Future<did_doc_delta>
```

Comment:
Does this persist the doc? Can this be called repeatedly?

Reply:
This just creates a Delta. No changes to a local copy of DID DOC are made.
How exactly the local DID DOC copy (if any) is changed is defined by a different layer (see the comment with 3 layers).